kratos/modeler/create_entities_from_geometries_modeler.cpp

```diff
@@ -0,0 +1,168 @@
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/parallel_environment.h"
+#include "utilities/entities_utilities.h"
+#include "utilities/variable_utils.h"
+#include "create_entities_from_geometries_modeler.h"
+
+namespace Kratos
+{
+
+namespace {
+
+/**
+ * @brief Create entities from geometries.
+ * @param EntityName the name of the entity
+ * @param rModelPart the model part
+ */
+template <class TEntitiesContainerType, class TEntityIdentifier>
+void CreateEntitiesFromGeometries(
+    const TEntityIdentifier& rEntityIdentifier,
+    ModelPart& rModelPart
+    )
+{
+    // Create the entities container and allocate space
+    TEntitiesContainerType entities_to_add;
+    entities_to_add.reserve(rModelPart.NumberOfGeometries());
+
+    // Get current max element id
+    using SizeType = std::size_t;
+    SizeType max_id;
+    const auto& r_root_model_part = rModelPart.GetRootModelPart();
+    if constexpr (std::is_same<typename TEntitiesContainerType::value_type, Element>::value) {
+        max_id = block_for_each<MaxReduction<SizeType>>(r_root_model_part.Elements(), [](auto& rElement){
+            return rElement.Id();
+        });
+    } else if constexpr (std::is_same<typename TEntitiesContainerType::value_type, Condition>::value) {
+        max_id = block_for_each<MaxReduction<SizeType>>(r_root_model_part.Conditions(), [](auto& rCondition){
+            return rCondition.Id();
+        });
+    }
+
+    // Loop geometries to create the corresponding entities from them
+    // Note that we retrieve the corresponding prototype entity from the entities idenfifier
+    // This makes possible to loop and create entities from different type geometries
+    for (auto& r_geom : rModelPart.Geometries()) {
+        if (rEntityIdentifier.HasPrototypeEntity(r_geom)) {
+            const auto& r_ref_entity = rEntityIdentifier.GetPrototypeEntity(r_geom);
+            auto p_entity = r_ref_entity.Create(++max_id, r_geom, nullptr);
+            entities_to_add.push_back(p_entity);
+        }
+    }
+
+    // Add the created entities to current submodelpart
+    if constexpr (std::is_same<typename TEntitiesContainerType::value_type, Element>::value) {
+        rModelPart.AddElements(entities_to_add.begin(), entities_to_add.end());
+    } else if constexpr (std::is_same<typename TEntitiesContainerType::value_type, Condition>::value) {
+        rModelPart.AddConditions(entities_to_add.begin(), entities_to_add.end());
+    }
+}
+
+}
+
+template <>
+void CreateEntitiesFromGeometriesModeler::RemoveModelPartEntities<Element>(ModelPart &rModelPart)
+{
+    const SizeType n_elements = rModelPart.NumberOfElements();
+    KRATOS_WARNING_IF("CreateEntitiesFromGeometriesModeler", n_elements != 0)
```

Comment:
Why is this a warning? Isn't this why the function was made for? (Maybe should be `KRATOS_INFO_IF`).
Imho warning should be if n_elements == 0.

Also you could merge both functions (see `LoopEntitiesList`)

Reply:
The idea is to inform the user that there are elements (entities) to be removed.